test/language/statements/class/elements/privatefield-on-proxy.js

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2019 Caio Lima. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Successfully access private field on Proxy objects
+esid: sec-getvalue
+info: |
+  GetValue(V)
+    ...
+    5. If IsPropertyReference(V), then
+      ...
+      b. If IsPrivateReference(V), then
+        i. Let env be the running execution context's PrivateNameEnvironment.
+        ii. Let field be ? ResolveBinding(GetReferencedName(V), env).
+        iii. Assert: field is a Private Name.
+        iv. Return ? PrivateFieldGet(field, base).
+  PrivateFieldGet(P, O)
+    1. Assert: P is a Private Name value.
+    2. If O is not an object, throw a TypeError exception.
+    3. Let entry be PrivateFieldFind(P, O).
+    4. If entry is empty, throw a TypeError exception.
+    5. Return entry.[[PrivateFieldValue]].
+includes: [compareArray.js]
+features: [class, class-fields-private, Proxy]
+---*/
+
+function assertArray(l, r) {
+  assert(compareArray(l, r), r);
+}
+
+let arr = [];
+	
+class ProxyBase {
+  constructor() {
+    return new Proxy(this, {
+      get: function (obj, prop) {
+        arr.push(prop);
+        return obj[prop];
+      }
+    });
+  }
+}
+
+class Test extends ProxyBase {
+  #f = 3;
+  method() {
+    return this.#f;
+  }
+}
+
+let t = new Test();
+let r = t.method();
+assert.sameValue(r, 3);
+
+assertArray(arr, ['method']);
```

Comment:
You don't need assertArray, you can use `assert.compareArray(l, r)` directly and it will handle a nice error message if necessary. Just preserve the included harness file (`compareArray.js`) and you have it.

Reply:
Oh nice.